### DISCUSSION TITLE
How to create exemplar and expose via PrometheusExporter?

### DISCUSSION THREAD

**Discussion by nemodev** [open-telemetry/opentelemetry-java-instrumentation#7564] [Q&A] [answered]
I am trying to make exemplars exported from my application's custom metrics, using PrometheusExporter.

I have an counter instance of `io.opentelemetry.api.metrics.LongCounter` type in my code and use `add(Long l, Attributes attributes, Context context)` method. I assume the `Context` parameter would include exemplar data for the metric, but I don't see exemplars from the exported metrics data. 

I checked using below command.
`curl -H 'Accept: application/openmetrics-text; version=1.0.0; charset=utf-8' http://localhost:6666/metrics`

I can see my custom metric values there, but it doesn't have exemplar information like `# {trace_id="...",span_id="..."}`.

I enabled OpenMetrics option in the prometheus exporter config.
```
exporters:
  prometheus:
    endpoint: 0.0.0.0:6666
    send_timestamps: true
    enable_open_metrics: true
    resource_to_telemetry_conversion:
      enabled: true
```

Am I missing something or doing incorrectly?

I found [an article about using `client_java` library from Prometheus project](https://linuxczar.net/blog/2022/01/17/java-spring-boot-prometheus-exemplars/), but the example is using Prometheus metric interface (`io.prometheus.client.Histogram`), not opentelemetry metrics API. If possible, I don't want to introduce additional dependency(`client_java`) in my application and stick to opentelemetry metrics interface.

I also found another person(@irizzant) asked [similar question in this channel](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4616) and he got solved his issue, but I don't know how he generated exemplar in his code.

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @nemodev ,

Currently the OTel Prometheus exporter only supports exemplars in histograms.
I'm not sure why we don't include exemplars when exporting other instruments' samples, I'll refer that question to @jack-berg

**Reply by jack-berg** [reply]
According to the spec, we're supposed to include exemplars on histograms and monotonic sums. We're currently not including them on monotonic sums (i.e. the aggregation of of counter instruments). I've opened [this issue](https://github.com/open-telemetry/opentelemetry-java/issues/5141) to track it.

**Reply by nemodev** [reply]
Hi @mateuszrzeszutek and @jack-berg ,

Thank you for the information. It is good to know that is current situation. I will wait for the counter metric support in the future.
